invenio_vocabularies/records/pidprovider.py

```diff
@@ -59,3 +59,61 @@ def create(cls, object_type=None, object_uuid=None, record=None, **kwargs):
             object_uuid=object_uuid,
             status=PIDStatus.REGISTERED
         )
+
+
+class CustomVocabularyPIDProvider(BaseProvider):
```

Comment:
need help: naming? 🤔 If we call this *Custom*, should we call the other one *Generic* i.e.

```diff
- VocabularyIdProvider
+ GenericVocabularyPIDProvider
```

Reply:
Yeah, not easy. In general for naming, it's good to think of what the class does and not for what it's used  - e.g. in this case it takes an id provided by the user so perhaps something like ``UserSpecifiedPIDProvider`` but I don't really like it.

Reply:
Yeah... tbh `CustomVocabularyPIDProvider` is the best I can come up with right now... :/ But I'm not convinced.